incubator/hnc/internal/reconcilers/object.go

```diff
@@ -332,11 +333,13 @@ func (r *ObjectReconciler) syncObject(ctx context.Context, log logr.Logger, inst
 
 	// Find the source object of the same name in the ancestors from top down to
 	// see if there's a conflicting source.
-	srcInst := r.Forest.Get(inst.GetNamespace()).GetSource(r.GVK, inst.GetName())
+	srcInst := r.Forest.Get(inst.GetNamespace()).GetAncestorSourceObject(r.GVK, inst.GetName())
 
 	// The object is a source without conflict if a copy of the source is not
-	// found in the forest or itself is found.
-	if srcInst == nil || srcInst.GetNamespace() == inst.GetNamespace() {
+	// found in the forest or itself is found. There won't be any conflict for
+	// types with sync modes other than 'Propagate' either.
+	mode := r.Forest.GetTypeSyncer(inst.GroupVersionKind()).GetMode()
```

Comment:
Can use `r.GVK` instead of `inst.GroupVersionKind()` here.

Reply:
Updated all occurrences.